src/garage/sampler/fragment_worker.py

```diff
@@ -0,0 +1,156 @@
+                 seed,
+                 max_path_length,
+                 worker_number,
+                 n_envs=DEFAULT_N_ENVS,
+                 timesteps_per_call=1):
+        super().__init__(seed=seed,
+                         max_path_length=max_path_length,
+                         worker_number=worker_number)
+        self._n_envs = n_envs
+        self._timesteps_per_call = timesteps_per_call
+        self._needs_env_reset = True
+        self._envs = [None] * n_envs
+        self._agents = [None] * n_envs
+        self._path_lengths = [0] * self._n_envs
+        self._complete_fragments = []
+        # Initialized in start_rollout
+        self._fragments = None
+
+    def update_env(self, env_update):
+        """Update the environments.
+
+        If passed a list (*inside* this list passed to the Sampler itself),
+        distributes the environments across the "vectorization" dimension.
+
+        Args:
+            env_update(gym.Env or EnvUpdate or None): The environment to
+                replace the existing env with. Note that other implementations
+                of `Worker` may take different types for this parameter.
+
+        Raises:
+            TypeError: If env_update is not one of the documented types.
+            ValueError: If the wrong number of updates is passed.
+
+        """
+        if isinstance(env_update, list):
+            if len(env_update) != self._n_envs:
+                raise ValueError('If separate environments are passed for '
+                                 'each worker, there must be exactly n_envs '
+                                 '({}) environments, but received {} '
+                                 'environments.'.format(
+                                     self._n_envs, len(env_update)))
+        elif env_update is not None:
+            env_update = [
+                copy.deepcopy(env_update) for _ in range(self._n_envs)
+            ]
+        if env_update:
+            for env_index, env_up in enumerate(env_update):
+                self._envs[env_index], up = _apply_env_update(
+                    self._envs[env_index], env_up)
+                self._needs_env_reset |= up
+
+    def start_rollout(self):
+        """Resets all agents if the environment was updated."""
+        if self._needs_env_reset:
+            self._needs_env_reset = False
+            self.agent.reset([True] * len(self._envs))
+            self._path_lengths = [0] * len(self._envs)
+            self._fragments = [InProgressTrajectory(env) for env in self._envs]
+
+    def step_rollout(self):
+        """Take a single time-step in the current rollout.
+
+        Returns:
+            bool: True iff at least one of the paths was completed.
+
+        """
+        prev_obs = np.asarray([frag.last_obs for frag in self._fragments])
+        actions, agent_infos = self.agent.get_actions(prev_obs)
+        completes = [False] * len(self._envs)
+        for i, action in enumerate(actions):
```

Comment:
why iterate over the number of actions? will this not always be 1 action?

Reply:
@krzentner bump

Reply:
as the PR description says, it's vectorized.